packages/auth/src/authClients/supabase.ts

```diff
@@ -66,21 +83,22 @@ export const supabase = (client: Supabase): AuthClientSupabase => {
     logout: async () => {
       return await client.auth.signOut()
     },
-    signup: async ({ email, password }) => {
-      return await client.auth.signUp({ email, password })
+    signup: async ({ email, password, redirectTo }) => {
+      return await client.auth.signUp({ email, password }, { redirectTo })
     },
-
     getToken: async () => {
       const currentSession = client.auth.session()
       return currentSession?.access_token || null
     },
     getUserMetadata: async () => {
       return await client.auth.user()
     },
-    // restore authentication when an OAuth or magiclink callback
-    // redirects back to site with access token
     restoreAuthState: async () => {
-      return await client.auth.refreshSession()
+      await client.auth.getSessionFromUrl()
+
+      window.history.replaceState({}, document.title, window.location.pathname)
```

Comment:
Are you not hitting bug #1892 with this code? If you are, I have a solution for that issue, just need to clean it up and get it out.

Reply:
It might be a similar issue - I'd have to look more into #1892 but that's for the Auth0 client and this code in the Supabase client.

`getSessionFromUrl()` is specific to the Supabase go-true client.